### ISSUE TITLE
Swapped (exchanged) labels in confusion matrix after classification training (Predicted <-> True)

### ISSUE THREAD

**Issue by HoLue** [ultralytics/ultralytics#3055]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

Training

### Bug

The labels in the confusion_matrix.png are wrong after classification training. "predicted" and "True" seem to be exchanged.
Here the confusion matrix after a mini training of mnist:
![confusion_matrix](https://github.com/ultralytics/ultralytics/assets/55573836/b6a55a10-cef5-46a8-b2e2-ec7eb564d010)

### Environment

- Ultralytics YOLOv8.0.114 🚀 Python-3.10.11 torch-2.0.1+cu118 CUDA:0 (Tesla T4, 15102MiB)
Setup complete ✅ (2 CPUs, 12.7 GB RAM, 23.3/166.8 GB disk)
- Colab notebook

### Minimal Reproducible Example

'''
# Pip install method (recommended)
%pip install ultralytics
import ultralytics
ultralytics.checks()

# Load YOLOv8n-cls, train it on mnist160
from ultralytics import YOLO

model_01 = YOLO('yolov8n-cls.pt')  # load a pretrained YOLOv8n classification model
model_01.train(data='mnist160', imgsz=32, epochs=3, fliplr=0.0)  # train the model
'''

### Additional

Here as zip the notebook YOLOv8_cls_bug02.ipynb: 
[YOLOv8_cls_bug02.zip](https://github.com/ultralytics/ultralytics/files/11666050/YOLOv8_cls_bug02.zip)

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@HoLue thank you for bringing this to our attention. We apologize for any inconvenience this issue may have caused. After carefully reviewing our codebase, we have identified and fixed the problem. The confusion matrix will now display the predicted labels under the predicted column, and the true labels under the true column, as expected. We appreciate your patience and feedback, as it helps us improve our software. Please let us know if you have any further questions or concerns.

**Comment by glenn-jocher** [comment]
@HoLue yes you are correct, there seems to be a bug in the classification confusion matrix. True and Predicted seem to be swapped.

**Comment by glenn-jocher** [comment]
@HoLue Great news 😃! Your original issue may now be resolved ✅ in PR #3241. This PR swaps the Classify confusion matrix axes, which were adding data incorrectly before (transposed).  

To get this update:
- **[Git](https://github.com/ultralytics/ultralytics)** – Run `git pull` from within your `ultralytics/` directory or run `git clone https://github.com/ultralytics/ultralytics` again
- **[Pip](https://pypi.org/project/ultralytics/)** – Update with `pip install -U ultralytics`
- **[Notebooks](https://github.com/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb)** – Check out the updated notebooks  <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov8"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **[Docker](https://hub.docker.com/r/ultralytics/ultralytics)** – Run `sudo docker pull ultralytics/ultralytics:latest` to update your image <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

Thank you for spotting this issue and letting us know. Please confirm if this update fixes the issue for you, and don't hesitate to report any other **issues** you find or **feature requests** you may have. Happy training with YOLOv8 🚀!
